os/board/common/partitions.c

```diff
@@ -122,18 +133,37 @@ void configure_partitions(void)
 
 #if defined(CONFIG_MTD_PARTITION_NAMES)
 		if (*names) {
-			mtd_setpartitionname(mtd_part, names);
-		}
-
-		while (*names) {
-			if (*names == ',') {
-				names++;
-				break;
+			while (*names) {
+				if (*names == ',') {
+					part_name[index] = '\0';
+					index = 0;
+					names++;
+					break;
+				}
+				if (index < MTD_PARTNAME_LEN) {
+					part_name[index++] = *(names++);
+				} else {
+					part_name[index] = '\0';
+					index = 0;
+					lldbg("ERROR: Partition name is so long. Please make it smaller than %d\n", MTD_PARTNAME_LEN);
+					while (*names) {
+						if (*(names++) == ',') {
+							break;
+						}
+					}
+					break;
+				}
+			}
+			mtd_setpartitionname(mtd_part, part_name);
+#ifdef CONFIG_BINARY_MANAGER
+			if (!strncmp(types, "bin,", 4)) {
+				binary_manager_register_bin_part(partno, part_name);
```

Comment:
Function definition has 3 parameters as below, and you are giving 2 parameters, Is it working fine?
void binary_manager_register_bin_part(int part_num, char *name, int partsize);